templates/base/head_opengraph.tmpl

```diff
@@ -0,0 +1,42 @@
+{{if .PageIsUserProfile}}
+	<meta property="og:title" content="{{.ContextUser.DisplayName}}">
+	<meta property="og:type" content="profile">
+	<meta property="og:image" content="{{.ContextUser.AvatarLink ctx}}">
+	<meta property="og:url" content="{{.ContextUser.HTMLURL}}">
+	{{if .ContextUser.Description}}
+		<meta property="og:description" content="{{.ContextUser.Description}}">
+	{{end}}
+{{else if .Repository}}
+	{{if .Issue}}
+		<meta property="og:title" content="{{.Issue.Title}}">
+		<meta property="og:url" content="{{.Issue.HTMLURL}}">
+		{{if .Issue.Content}}
+			<meta property="og:description" content="{{.Issue.Content}}">
+		{{end}}
+	{{else if or .PageIsDiff .IsViewFile}}
+		<meta property="og:title" content="{{.Title}}">
+		<meta property="og:url" content="{{AppUrl}}{{.Link}}">
```

Comment:
According to the standard, https://ogp.me/

`og:url - The canonical URL of your object that will be used as its permanent ID in the graph,`

But `.Link` is not the canonical URL , it may contain various different parameters.

Reply:
`.Link` does not contain parameters, that's `.CurrentURL`.
https://github.com/go-gitea/gitea/blob/79f73299710fe8596b6ea31c0fa0ab97ab3dc801/modules/context/context.go#L124
https://github.com/go-gitea/gitea/blob/79f73299710fe8596b6ea31c0fa0ab97ab3dc801/modules/context/context.go#L158-L159